test/unity/toolkits/object_detection/one_shot_object_detection/test_parameter_sampler.cxx

```diff
@@ -0,0 +1,105 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define BOOST_TEST_MODULE test_parameter_sampler
+
+#include <toolkits/object_detection/one_shot_object_detection/util/parameter_sampler.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include <core/util/test_macros.hpp>
+
+#include <unordered_map>
+
+static constexpr int max_dimension = 6000;
+static constexpr int num_rows = 1000;
+
+namespace turi {
+namespace one_shot_object_detection {
+namespace {
+
+ParameterSampler create_parameter_sampler(std::default_random_engine &engine) {
+
+  std::uniform_int_distribution<int> dist(0, max_dimension);
+  
+  size_t starter_width = dist(engine);
+  size_t starter_height = dist(engine);
+  size_t dx = 0;
+  size_t dy = 0;
+  ParameterSampler sampler(starter_width, starter_height, dx, dy);
+  return sampler;
+}
+
+std::unordered_map<std::string, std::vector<double> > get_all_angles(
+    ParameterSampler &sampler, int seed) {
+  std::unordered_map<std::string, std::vector<double> > all_angles;
+  std::vector<double> thetas;
+  std::vector<double> phis;
+  std::vector<double> gammas;
+  for (int i=0; i<num_rows; i++) {
+    sampler.sample(seed+i);
+    double theta = sampler.get_theta();
+    double phi = sampler.get_phi();
+    double gamma = sampler.get_gamma();
+    thetas.push_back(theta);
+    phis.push_back(phi);
+    gammas.push_back(gamma);
+  }
+  all_angles["theta"] = thetas;
+  all_angles["phi"] = phis;
+  all_angles["gamma"] = gammas;
+  return all_angles;
+}
+
+int count_in_range(std::vector<double> samples, double lower, double upper) {
+  int count = 0;
+  for (double sample : samples) {
+    if (sample >= lower && sample <= upper) {
+      count += 1;
+    }
+  }
+  return count;
+}
+
+bool angles_match_distribution(std::vector<double> angles,
```

Comment:
It's testing code, so less important, but still a good idea to get used to the `const type&` pattern.

Reply:
Done.